### DISCUSSION TITLE
Caching in .NET with MemoryCache

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#50] [General]
# Caching in .NET with MemoryCache

In this blog post, we will discuss how we can "cache" entries from the database. We will talk about why we would do this in the first place and how to achieve that.
Also, we will talk about some implications and what "cache invalidation" is.

https://steven-giesel.com/blogPost/b355a042-2c5b-4d96-b1ca-a9dc0fc23509

**Comment by Virslandir** [comment]
Hi Steven, great post. I just wanted to point out that in the Expiration window section, right above the first image, these two sentences should probably be a single sentence:

To overcome that the IMemoryCache offers us an overload where we can. Pass in some additional options.

**Reply by linkdotnet** [reply]
Thanks, and yes absolutely. It should be fixed now!

**Comment by ghadzhigeorgiev** [comment]
Great article, thanks!

**Comment by victorsigga** [comment]
Great article. Could you give an example of how we can use it in .net maui. How to declare in maui program, and how we would call it to use in code

**Reply by linkdotnet** [reply]
I don't have much exposure to .NET maui myself. I am not sure if it is directly included, if not you can add the package like described in the article.

Still the same principles hold true as shown. You can cache certain data you would retrieve from a backend on a page load.
So instead of everytime going to said page (under the condition the data doesn't change often) you save it in the cache so that the next page load is quicker.

**Comment by msgritsenko** [comment]
Hello. I assume we should prevent several calls to backend if several requests came with the same Id while there is no caching data.
I used a ConcurrentDictionary<Id, Task<Data>> to prevent new calls to backend. We need recall only if there is no key in cache and dictionary.

**Reply by linkdotnet** [reply]
Hey @msgritsenko. It depends a bit on your specific needs:

For this blog for example I do not expect 2 simultaneous requests. Furthermore, even if there are two requests hitting at the same time without the cache being populated, it doesn't hurt much performance-wise if I go twice to the database.

It makes the code much smaller with those "requirements".
